src/meta/service/src/stream_service.rs

```diff
@@ -461,4 +461,35 @@ impl StreamManagerService for StreamServiceImpl {
             .await?;
         Ok(Response::new(ListRateLimitsResponse { rate_limits }))
     }
+
+    async fn alter_sink_config(
+        &self,
+        request: Request<AlterSinkConfigRequest>,
+    ) -> Result<Response<AlterSinkConfigResponse>, Status> {
+        let request = request.into_inner();
+
+        let actor_to_apply = self
+            .metadata_manager
+            .update_sink_config_by_sink_id(request.sink_id as i32, request.config.clone())
```

Comment:
We should't do the update sink config here. We should apply the config to meta store in `CommandContext::post_collect` after we successfully collect the mutation barrier.

Reply:
This approach is acceptable to me, a failed barrier is likely to trigger a recover.
IIRC, we also update metadata first and then push a mutation when `alter rate limit`.

Reply:
Got it, then I'm okay with it.